camel-idea-plugin/src/main/java/org/apache/camel/idea/util/CamelIdeaUtils.java

```diff
@@ -168,7 +168,7 @@ public static boolean isCameSimpleExpressionUsedAsPredicate(PsiElement element)
                         element = element.getParent();
                     }
                     if (element instanceof PsiMethodCallExpression) {
-                        exp = (PsiMethodCallExpression) element;
+                        exp = PsiTreeUtil.getParentOfType(element.getParent(), PsiMethodCallExpression.class);
```

Comment:
previous it didn't go far enough back in the PSI tree to find parent method and therefore it didn't see the simple method was an input predicate to another method.